### PR TITLE
Runtime component for Map type downlinks.

### PR DESCRIPTION
**Pull request by horned-sphere**
Follows on from #483.

### CODE REVIEW

File: runtime/swim_runtime/src/downlink/failure.rs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+use std::{convert::Infallible, fmt::Display, num::NonZeroUsize};
+
+use swim_utilities::format::comma_sep;
+use thiserror::Error;
+use tracing::warn;
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum BadFrameResponse<E> {
+    /// Instruction ignore the bad envelope.
```
Comment (DobromirM) [inline runtime/swim_runtime/src/downlink/failure.rs:23]:
```suggestion
    /// Instruction to ignore the bad envelope.
```

**Review by DobromirM — APPROVED** [review]

File: runtime/swim_runtime/src/downlink/interpretation/mod.rs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+use swim_api::protocol::map::{
+    extract_header, MapMessageEncoder, RawMapOperation, RawMapOperationEncoder,
+};
+use swim_recon::parser::MessageExtractError;
+use tokio_util::codec::Encoder;
+
+#[cfg(test)]
+mod tests;
+
+/// A possible transformation to apply to an incoming event body, before passing it on
```
Comment (SirCipher) [inline runtime/swim_runtime/src/downlink/interpretation/mod.rs:27]: Why possibly? Is the transformation only sometimes applied?

File: runtime/swim_runtime/src/downlink/interpretation/mod.rs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+/// to the downlink implementation.
+pub trait DownlinkInterpretation {
+    type Error;
+
+    const SINGLE_FRAME_STATE: bool = true;
```
Comment (SirCipher) [inline runtime/swim_runtime/src/downlink/interpretation/mod.rs:32]: What is this constant?

File: runtime/swim_runtime/src/downlink/interpretation/mod.rs (hunk 2 above)
Comment (horned-sphere) [reply runtime/swim_runtime/src/downlink/interpretation/mod.rs:27]: In the case of a value downlink, it just gets passed through as is. For map downlinks, it gets decomposed into the key and value components first.

File: runtime/swim_runtime/src/downlink/interpretation/mod.rs (hunk 3 above)
Comment (horned-sphere) [reply runtime/swim_runtime/src/downlink/interpretation/mod.rs:32]: It dermines of a single frame can provide the entire state for the downlink (value like) or whether multiple frames might be required (map like). I'll add some documentation.

**Review by SirCipher — APPROVED** [review]

**Pull request merged**